pkg/schema/searchComplete.go

```diff
@@ -2,61 +2,104 @@ package schema
 
 import (
 	"context"
-	"strconv"
 
+	"github.com/doug-martin/goqu/v9"
+	"github.com/doug-martin/goqu/v9/exp"
+	"github.com/driftprogramming/pgxpoolmock"
 	"github.com/stolostron/search-v2-api/graph/model"
+	"github.com/stolostron/search-v2-api/pkg/config"
 	db "github.com/stolostron/search-v2-api/pkg/database"
+	res "github.com/stolostron/search-v2-api/pkg/resolver"
 	klog "k8s.io/klog/v2"
 )
 
+type SearchCompleteResult struct {
+	input    *model.SearchInput
+	pool     pgxpoolmock.PgxPool
+	property string
+	limit    *int
+	query    string
+	params   []interface{}
+}
+
+func (s *SearchCompleteResult) autoComplete(ctx context.Context) ([]*string, error) {
+	s.searchCompleteQuery(ctx)
+	res, autoCompleteErr := s.searchCompleteResults()
+	if autoCompleteErr != nil {
+		klog.Error("Error resolving properties in autoComplete", autoCompleteErr)
+	}
+	return res, autoCompleteErr
+}
+
 func SearchComplete(ctx context.Context, property string, srchInput *model.SearchInput, limit *int) ([]*string, error) {
-	query := searchCompleteQuery(ctx, property, srchInput, limit)
-	klog.Infof("SearchComplete Query: ", query)
-	return searchCompleteResults(query)
+
+	searchCompleteResult := &SearchCompleteResult{
+		input:    srchInput,
+		pool:     db.GetConnection(),
+		property: property,
+		limit:    limit,
+	}
+	return searchCompleteResult.autoComplete(ctx)
+
 }
 
-func searchCompleteQuery(ctx context.Context, property string, input *model.SearchInput, limit *int) string {
-	var selectClause, limitClause, limitStr, query string
-	if property != "" {
-		klog.Infof("property: %s and limit:%d", property, limit)
-		if property == "cluster" {
-			//Adding WHERE clause to filter out NULL values and ORDER by sort results
-			selectClause = "SELECT DISTINCT " + property + " FROM search.resources WHERE " + property + " IS NOT NULL ORDER BY " + property
+func (s *SearchCompleteResult) searchCompleteQuery(ctx context.Context) {
+	var limit int
+	var whereDs []exp.Expression
+	var selectDs *goqu.SelectDataset
+
+	//FROM CLAUSE
+	schemaTable := goqu.S("search").Table("resources")
+	ds := goqu.From(schemaTable)
+	//WHERE CLAUSE
+	if s.input != nil && len(s.input.Filters) > 0 {
+		whereDs = res.WhereClauseFilter(s.input)
+	}
+	//SELECT CLAUSE
+	if s.property != "" {
```

Comment:
We could move this `if` to the start of this function.  If we don't have this value we can't do anything.

Reply:
Done